modules/programs/brew-bundle.nix

```diff
@@ -0,0 +1,225 @@
+    " brew bundle --file='${brewfile}' --no-lock" +
+    (if cfg.cleanupType == "uninstall" || cfg.cleanupType == "zap" then " --cleanup" else "") +
+    (if cfg.cleanupType == "zap" then " --zap" else "");
+
+in
+
+{
+  options.programs.brew-bundle = {
+    enable = mkEnableOption ''
+      configuring your Brewfile, and installing/updating the formulas therein via
+      the <command>brew bundle</command> command, using <command>nix-darwin</command>.
+
+      Note that enabling this option does not install Homebrew. See the Homebrew website for
+      installation instructions: https://brew.sh
+    '';
+
+    noAutoUpdate = mkOption {
+      type = types.bool;
+      default = true;
+      example = false;
+      description = ''
+        Sets the <literal>HOMEBREW_NO_AUTO_UPDATE</literal> environment variable when running the
+        <command>brew bundle</command> command. The default is <literal>true</literal> so that
+        repeated invocations of <command>darwin-rebuild switch</command> are idempotent.
+      '';
+    };
+
+    cleanupType = mkOption {
+      type = types.enum [ "none" "uninstall" "zap" ];
+      default = "none";
+      example = "uninstall";
+      description = ''
+        This option manages what happens to formulas installed by Homebrew, that aren't present in
+        the Brewfile generated by this module.
+
+        When set to <literal>"none"</literal> (the default), formulas not present in the generated
+        Brewfile are left installed.
+
+        When set to <literal>"uninstall"</literal>, <command>nix-darwin</command> invokes
+        <command>brew bundle install</command> with the <command>--cleanup</command> flag. This
+        uninstalls all formulas not listed in generate Brewfile, i.e.,
+        <command>brew uninstall</command> is run for those formulas.
+
+        When set to <literal>"zap"</literal>, <command>nix-darwin</command> invokes
+        <command>brew bundle install</command> with the <command>--cleanup --zap</command>
+        flags. This uninstalls all forumalas not listed in the generated Brewfile, and if the
+        formula is a cask, removes all files associated with the cask. In other words,
+        <command>brew uninstall --zap</command> is run for all those formulas.
+
+        If you plan on exclusively using <command>nix-darwin</command> to manage formulas installed
+        by Homebrew, you probably want to set this option to <literal>"uninstall"</literal> or
+        <literal>"zap"</literal>.
+      '';
+    };
+
+    setNoLockEnvvar = mkOption {
+      type = types.bool;
+      default = true;
+      description = ''
+        Sets the <literal>HOMEBREW_BUNDLE_NO_LOCK</literal> enviroment variable, by adding it to
+        <option>environment.variables</option>, so that lock files aren't generated when/if you run
+        the <command>brew bundle</command> command yourself.
+      '';
+    };
+
+    setBrewfileEnvvar = mkOption {
```

Comment:
hmmm.  For naming this, do you use this option, and if so, why?  I'm only coming up with names like `globalConfig` or something, which is super unclear.

Reply:
Yeah, I do use it. My main use case is that there are times where I want to manually invoke `brew bundle` to, e.g., run a cleanup after I've installed some apps etc. manually with `brew` that I wanted to test or whatever, or update apps specified in my `Brewfile` generated by this module (without having to run a full `darwin-rebuild switch`). With this option enabled, I can do that seamlessly without having to explicitly reference the Brewfile in the store (which is annoying to do).

Reply:
As mention in another comment, I renamed this option to `userConfig.brewfile`.

Reply:
Resolving this comment thread, and picking the conversation up about these options, in a new thread.